core/src/main/java/tc/oc/pgm/consumable/ConsumableModule.java

```diff
@@ -0,0 +1,58 @@
+package tc.oc.pgm.consumable;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Logger;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.kits.Kit;
+import tc.oc.pgm.kits.KitNode;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.Node;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class ConsumableModule implements MapModule<ConsumableMatchModule> {
+
+  private Set<ConsumableDefinition> consumableDefinitions = new HashSet<>();
+
+  @Override
+  public @Nullable ConsumableMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    return new ConsumableMatchModule(match, consumableDefinitions);
+  }
+
+  public static class Factory implements MapModuleFactory<ConsumableModule> {
+
+    @Override
+    public @Nullable ConsumableModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      ConsumableModule consumableModule = new ConsumableModule();
+
+      for (Element consumableElement :
+          XMLUtils.flattenElements(doc.getRootElement(), "consumables", "consumable")) {
+        String id = XMLUtils.getRequiredAttribute(consumableElement, "id").getValue();
+        boolean override = XMLUtils.parseBoolean(consumableElement.getAttribute("override"), true);
+        Kit kit = factory.getKits().parseKitProperty(consumableElement, "kit", KitNode.EMPTY);
```

Comment:
You can look at shops module for an up-to-date example on how to parse a player action instead of a kit, generally speaking it will change very little, however make sure that you define the dependency between modules

Reply:
Not 100% sure I got the dependencies right but I think I did.